Source/Core/VideoBackends/D3D/GeometryShaderCache.cpp

```diff
@@ -228,7 +221,7 @@ bool GeometryShaderCache::SetShader(u32 primitive_type)
 	}
 
 	// Need to compile a new shader
-	ShaderCode code = GenerateGeometryShaderCode(primitive_type, API_D3D);
+	ShaderCode code = GenerateGeometryShaderCode(primitive_type, API_D3D, uid.GetUidData());
```

Comment:
May you include (or likely use) the primitive_type of the uid itself? This would kill another reference. Same for dstAlphaMode of pixelshaders.